### PR TITLE
Creation of ZendesK

### PR DESCRIPTION
**Pull request by JeffNaef-Triggermesh**
keeping the user inside the triggermesh interface and registering the integration has proved to be a little difficult (at least for me) You can find all of this logic in the reconciler.go. I still have to finish creating the proper query to register the Zendesk registration Trigger (because the documentation is a bit spotty on that part) but it does create a Zendesk Extension for you on deployment reconciliation with values passed in via the Spec. There is a little bit of developer notes left in here because this was not only my first Source but also the first time I have done anything in the Reconciler... so there are a few "rough" corners and I apologize for this.

### CODE REVIEW

File: zendesk/sample/300-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: zendesksource
+type: Opaque
+stringData:
+  token: '' #Example: 5LCHYusFc7r2Ayb97gbVY4svUgPM07x7Erep1yx
+  password: '' #Example: Mypassword
```
Comment (cab105) [inline zendesk/sample/300-secret.yaml:22]: Are both required now? Thought this would just be the API token.

File: zendesk/cmd/adapter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM golang:1.14-stretch AS builder
```
Comment (cab105) [inline zendesk/cmd/adapter/Dockerfile:1]: Should have a copyright notice

File: zendesk/cmd/controller/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
+FROM golang:1.14-stretch AS builder
```
Comment (cab105) [inline zendesk/cmd/controller/Dockerfile:1]: Should have a copyright notice here.

File: zendesk/sample/300-secret.yaml (hunk 1 above)
Comment (JeffNaef-Triggermesh) [reply zendesk/sample/300-secret.yaml:22]: for the source we need 
A Token: to create the webhooks from within the source
A Password and username: To define for the basic HTTP Auth.

Comment (JeffNaef-Triggermesh) [reply zendesk/sample/300-secret.yaml:22]: Oh okay! I will get rid of them!

**Pull request closed without merging**